### PR TITLE
Use System specific line separator

### LINKED ISSUE #1539: XtextDocumentModifyTest / XtextGrammarQuickfixTest fails on windows
XtextDocumentModifyTest / XtextGrammarQuickfixTest fails on windows

i assume that the (test) api is line ending dependent somehow

XtextGrammarQuickfixTest actually seems to be flaky
maybe https://github.com/eclipse/xtext-eclipse/issues/1022 fix is not properly working
as also the label is same?

```

Failures:
  XtextGrammarQuickfixTest.test_convert_terminal_fragment_to_terminal_rule:85->AbstractQuickfixTest.testQuickfixesOn:176->AbstractQuickfixTest.quickfixesAreOffered:217->AbstractQuickfixTest.assertIssueResolutionResult:259->Assert.assertEquals:146->Assert.assertEquals:117 expected:<...example/mydsl/MyDsl"[

]odel hidden():
        a = ID;

termina...> but was:<...example/mydsl/MyDsl"[

Model hidden():]
        a = ID;

termina...>
  XtextGrammarQuickfixTest.test_fix_invalid_hidden_token:171->AbstractQuickfixTest.testQuickfixesOn:176->AbstractQuickfixTest.quickfixesAreOffered:217->AbstractQuickfixTest.assertIssueResolutionResult:259->Assert.assertEquals:146->Assert.assertEquals:117 expected:<...example/mydsl/MyDsl"[

]odel hidden():
        a = ID;

Another...> but was:<...example/mydsl/MyDsl"[

Model hidden():]
        a = ID;

Another...>

Tests run: 265, Failures: 2, Errors: 0, Skipped: 1
````

### PR DESCRIPTION
**Pull request by cdietrich**
Use System specific line separator
Fixes #1539

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProviderTest.java [hunk 1]
```diff
@@ -333,7 +333,7 @@ private String grammarGeneratingModel(String... bodyContent) {
 		list.add("grammar org.xtext.example.mydsl.MyDsl with org.eclipse.xtext.common.Terminals");
 		list.add("generate myDsl \"http://www.xtext.org/example/mydsl/MyDsl\"");
 		list.addAll(Arrays.asList(bodyContent));
-		return Strings.concat("\n", list);
+		return Strings.concat(System.lineSeparator(), list);
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProviderTest.java:336]: Should this be done for the other lines in this file, too, that use `Strings.concat` ?

**Review by szarnekow — APPROVED** [review]

File: org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProviderTest.java (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProviderTest.java:336]: there are more?

Comment (szarnekow) [reply org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProviderTest.java:336]: e.g. line 82

Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProviderTest.java:336]: good catch. updated

**Review by szarnekow — APPROVED** [review]

**Pull request merged**